### ISSUE TITLE
Inconsistent Results

### ISSUE THREAD

**Issue by arkihillel** [james-proxy/james-browser-launcher#68]
Each time I run a `detect` I get different results.

Running the module on OS X Sierra with nodeJS 8.4.0 with a couple of external hard drives

I get from 1 to 8 results depending on the run

**Comment by mitchhentges** [comment]
Are any results consistent? When you get 1 result, is it always the same one?
Thanks for the report, I'll look into this later this week

**Comment by arkihillel** [comment]
Thank you!

From my last tests and as far as I can remember Chromium is always
detected. Safari Chrome Firefox and Opera appear randomly and sometimes
only from the versions I have on my external hard drive

I should add that I did not try to run the module with my external disks
unplugged

**Comment by mitchhentges** [comment]
Oh, there's some installations on the external drives, gotcha.

Yeah, if you can try running the module with the externals uplugged, that'd be radical. Also, just to see if there's a pattern: can you list:

Browser|Location of browser (external drive/main drive)|Is consistently listed/consistently unlisted/intermitten by the module
------|-----|-----

**Comment by arkihillel** [comment]
I did a fresh install of High Sierra yesterday, looks like the results are a bit different now and I can't reproduce the inconsistent number of browsers detected. 

After 10 runs without external volumes:
All the browsers are detected perfectly now 👍 

After 10 runs with external volumes:
I consistently get Chrome, Safari, Firefox, Chromium and Opera. They are coming from different sources though:

e.g.

``` 
/Volumes/Western Digital/backup/Applications/Google Chrome.app
/Applications/Firefox.app
/Volumes/Carbon Copy/Applications/Chromium.app
/Volumes/Western Digital/backup/Applications/Safari.app
/Volumes/Carbon Copy/Applications/Opera.app
``` 
``` 
/Volumes/Western Digital/backup/Applications/Google Chrome.app
/Volumes/Carbon Copy/Applications/Firefox.app
/Volumes/Carbon Copy/Applications/Chromium.app
/Volumes/Carbon Copy/Applications/Opera.app
/Volumes/Carbon Copy/Applications/Safari.app
``` 
``` 
/Volumes/Carbon Copy/Applications/Chromium.app
/Applications/Firefox.app
/Volumes/Western Digital/backup/Applications/Safari.app
/Applications/Opera.app
/Volumes/Western Digital/backup/Applications/Google Chrome.app
```
 
I didn't go through the sources but I guess that you're only expected one instance of each supported browser which kinda makes sense if most cases but doesn't really work in this case

Providing an option to ignore external volumes would be make it work well enough for me. Although I would still deem this a bug

**Comment by mitchhentges** [comment]
Ah, I understand.
How do you have these browsers installed in two different locations? Is `/Applications/$BROWSER.app` a symlink to `/Volumnes/...`? Do you have two different versions of the same browser installed at once?

**Comment by arkihillel** [comment]
Not exactly, the other volumes just contain backups which happen to have (often old) versions of the same browsers.
